player/command.c

```diff
@@ -6409,20 +6421,12 @@ static void cmd_track_reload(void *p)
 
     if (nt_num < 0) {
         cmd->success = false;
+        talloc_free(lang);
         return;
     }
 
     struct track *nt = mpctx->tracks[nt_num];
-
-    if (!nt->lang) {
-        enum track_flags flags = 0;
-        bstr lang = mp_guess_lang_from_filename(bstr0(nt->external_filename), NULL,
-                                                &flags);
-        nt->lang = bstrto0(nt, lang);
-        nt->hearing_impaired_track = flags & TRACK_HEARING_IMPAIRED;
-        nt->forced_track = flags & TRACK_FORCED;
-        nt->default_track = flags & TRACK_DEFAULT;
-    }
+    nt->lang = talloc_steal(nt, lang);
```

Comment:
Could you explain what language is always override now from old track? Previously it was only done if lang was not set.

Reply:
Because I added that condition based on the wrong assumption that external tracks can have language metadata. Well technically you can do something like `sub-add foo.mkv` and change the sub language before reloading though that is extremely uncommon. So we can alternatively do:

```c
if (nt->lang) {
    talloc_free(lang);
} else {
    nt->lang = talloc_strdup(nt, lang);
}
```

This will reflect language changes in `foo.mkv` but will be incorrect in the original case I mentioned of reomving the track language before reloading. But surely calling `sub-add` with a language is more common so it's better to fix that.

Reply:
Ok so actually `sub-reload` with an mkv file adds the video and audio tracks again everytime but not the sub track. So I updated as above in order to use any new language. Though reloading doesn't work properly in this case anyway since it keeps adding unrelated tracks.